TWLight/applications/helpers.py

```diff
@@ -170,38 +171,71 @@ def get_output_for_application(app):
 
 
 def get_active_authorizations(partner_pk, stream_pk=None):
-    '''
-    Retrieves the numbers of active authorizations available for a particular
+    """
+    Retrieves the active authorizations available for a particular
     partner or collections if stream_pk is not None. Active authorizations are
-    authorizations having expiry dates greater than today.
-    '''
+    authorizations with which we can operate, and is decided by certain conditions as
+    spelled out in the is_valid property of the Authorization model object (users/models.py).
+    """
     today = datetime.date.today()
     try:
-        if stream_pk is None:
-            active_authorizations = Authorization.objects.filter(date_expires__gt=today, partner=partner_pk)
-        else:
-            active_authorizations = Authorization.objects.filter(date_expires__gt=today, partner=partner_pk, stream=stream_pk)
-        return active_authorizations.count()
+        # The filter below is equivalent to retrieving all authorizations for a partner
+        # and (or) stream and checking every authorization against the is_valid property
+        # of the authorization model, and hence *must* be kept in sync with the logic in
+        # TWLight.users.model.Authorization.is_valid property.
+        active_authorizations = Authorization.objects.filter(
+            Q(date_expires__isnull=False, date_expires__gte=today) |
+            Q(date_expires__isnull=True),
+            authorizer__isnull=False,
+            user__isnull=False,
+            partner_id__isnull=False,
+            date_authorized__isnull=False,
+            date_authorized__lte=today,
+            partner=partner_pk
+        )
+        if stream_pk:
+            active_authorizations = active_authorizations.filter(stream=stream_pk)
+
+        return active_authorizations
     except Authorization.DoesNotExist:
```

Comment:
Is this an exception that we actually hit? If so return `Authorization.objects.none()` here to return an empty Authorization queryset that way you can always call queryset methods on the return value.